### ISSUE TITLE
Feature request: Input Hysteresis

### ISSUE THREAD

**Issue by vzorglub** [cflurin/node-red-contrib-ramp-thermostat#11]
Hello,
Would it be possible to add an input topic, eg: setHysteresis
to be able to modify the hysteresis on the fly

Thanks

**Comment by cflurin** [comment]
@vzorglub
I'm working on an urgent project just now but I'll have a look at your request as soon as possible.

**Comment by cflurin** [comment]
@vzorglub 
Could you show a practical use case for this feature?

**Comment by vzorglub** [comment]
So that a user can change the hysteresis if there is too much triggering. The user can do this on a ui. Without going into node-red itself. Application for non computer users and WAF factor.

**Comment by cflurin** [comment]
I see.
However one thing you should consider is that all the run-time settings will be lost after a node-red restart and also on deploy (except the modes "modified flows", "modified nodes" if the ramp-thermostat node isn't modified).
Therefore you have to save the settings in a file or using MQTT and set them on start/deploy.

**Comment by vzorglub** [comment]
Yes,
Thank you
I am aware of this have persistence setup in my flows
I can deploy to users. I have tried to modify your code but I am new at
Javascript and it doesn't work. So I am looking forward to see your code.

Thanks

On 3 Jan 2018 18:12, "cflurin" <notifications@github.com> wrote:

I see.
However one thing you should consider is that all the run-time settings
will be lost after a node-red restart and also on deploy (except the modes
"modified flows", "modified nodes" if the ramp-thermostat node isn't
modified).
Therefore you have to save the settings in a file or using MQTT and set
them on start/deploy.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/cflurin/node-red-contrib-ramp-thermostat/issues/11#issuecomment-355084204>,
or mute the thread

**Comment by cflurin** [comment]
I've just uploaded a new version.
Please install from GitHub (you can also only download `ramp-thermostat.js` if you like).

howto:

```sh
msg.topic: setHysteresisPlus
msg.payload: nn.n (number)
```
and
```sh
msg.topic: setHysteresisMinus
msg.payload: nn.n (number)
```

**Comment by cflurin** [comment]
@vzorglub 
Did you test this feature?
Your feedback is welcome.

**Comment by vzorglub** [comment]
Sorry
Not yet
But I saw the code.
Now I get it!
Thanks

**Comment by maragelis** [comment]
Was going to work on this, @cflurin thanks just what I needed to complete my smart thermostat.

**Comment by cflurin** [comment]
@maragelis if this feature works for you, I'll expand the documentation and upload a new version.

**Comment by vzorglub** [comment]
Hello,
I have tested it and managed to tune the heating remotely with it rather
than have to log in and change the nodes configurations.  So, please add it
to your distribution. Thanks
Vincent

**Comment by cflurin** [comment]
Hi,
I've published Version 0.5.0.
It would be helpful for other users if you could upload an example to the wiki.
Thanks.
